pkgs/development/web/mailcatcher/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, lib, bundlerEnv, ruby, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "mailcatcher-${version}";
+
+  version = (import ./gemset.nix).mailcatcher.version;
+
+  env = bundlerEnv {
+    name = "${name}-gems";
+
+    inherit ruby;
+
+    gemdir = ./.;
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  phases = ["installPhase"];
```

Comment:
The actual ruby packaging work is done by bundlerEnv, the wrapper is needed to avoid exposing `bundler`, `thin` and `rackup` to the user.
I applied the reasoning found here: http://blog.arkency.com/2016/04/packaging-ruby-programs-in-nixos/